src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryExtensions.cs

```diff
@@ -23,7 +24,7 @@ public static void Append(this IHeaderDictionary headers, string key, StringValu
         /// </summary>
         /// <param name="key">The header name.</param>
         /// <param name="values">The header values.</param>
-        public static void AppendCommaSeparatedValues(this IHeaderDictionary headers, string key, params string[] values)
+        public static void AppendCommaSeparatedValues(this IDictionary<string, StringValues> headers, string key, params string[] values)
```

Comment:
These CommaSeparatedValues methods are actually very specific to request/response header notation... That is, for the header `x-example: "detect, commas", "inside", values, "like this"` and turn that into the string array: 
* `detect, commas`
* `inside`
* `values`
* `like this`

That is not a behavior which is appropriate for cookies, query, or form values - so the extension method on the common interface should be avoided...

What if `public interface IHeaderDictionary : IDictionary<string, StringValues> {}` was added to the same assembly as `IHttpRequest/ResponseFeature`? Doesn't add extra implementation burden to the server (just need to add that interface to the dictionary impl), you can put extension methods on it specifically, and you don't need to allocate a wrapper between the http abstraction class and the feature interface